compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -71,6 +71,10 @@ object ProtoTypes {
                    |constraint was: ${ctx.typerState.constraint}
                    |constraint now: ${newctx.typerState.constraint}""")
             if result && (ctx.typerState.constraint ne newctx.typerState.constraint) then
+              val tvars = (newctx.typerState.ownedVars -- ctx.typerState.ownedVars).toList
```

Comment:
Isn't this equal to `newctx.typerState.ownedVars` ? Otherwise the assert in https://github.com/scala/scala3/blob/eb5c3e81304eee5ce75c2796bd4b7b16d2d1b19b/compiler/src/dotty/tools/dotc/core/TyperState.scala#L288-L289 should trigger

Reply:
This was meant to be the shorthand for `newctx.typerState.ownedVars -- locked`, with a `val locked = newctx.typerState.ownedVars` before testCompat.  But I'll doublecheck what you're pointing to.

Reply:
Yes, you're right, will simplify.